lib/ain-ocean/src/indexer/loan_token.rs

```diff
@@ -0,0 +1,189 @@
+                };
+                active_price = OraclePriceActiveActive {
+                    amount: previous_price[0].active.amount.clone(),
+                    weightage: previous_price[0].active.weightage,
+                    oracles: oracles,
+                };
+            }
+
+            let oracle_price_active = OraclePriceActive {
+                id: price_active_id.clone(),
+                key: oracle_price_key,
+                sort: hex::encode(ctx.block.height.to_be_bytes()),
+                active: active_price.clone(),
+                next: next_price.clone(),
+                is_live: is_live(Some(active_price), Some(next_price)),
+                block: ctx.block.clone(),
+            };
+            services
+                .oracle_price_active
+                .by_id
+                .put(&price_active_id, &oracle_price_active)?;
+            services
+                .oracle_price_active
+                .by_key
+                .put(&oracle_price_active.key, &oracle_price_active.id)?;
+        }
+
+        Ok(())
+    }
+    fn invalidate(&self, services: &Arc<Services>, context: &Context) -> Result<()> {
+        let ticker_id = (
+            self.currency_pair.token.clone(),
+            self.currency_pair.currency.clone(),
+            context.block.height,
+        );
+        services.oracle_price_active.by_id.delete(&ticker_id)?;
+        services
+            .oracle_price_active
+            .by_key
+            .delete(&(ticker_id.0, ticker_id.1))?;
+        Ok(())
+    }
+}
+pub fn aggregated_validate(aggrigated_price: OraclePriceAggregated, context: &Context) -> bool {
+    let minimum_live_oracles = 2;
+    if (aggrigated_price.block.time - context.block.time).abs() >= 3600 {
+        return false;
+    }
+    if aggrigated_price.aggregated.oracles.active < minimum_live_oracles {
+        return false;
+    }
+
+    if aggrigated_price.aggregated.weightage <= 0 {
+        return false;
+    }
+
+    true
+}
+
+pub fn is_live(
+    active: Option<OraclePriceActiveActive>,
+    next: Option<OraclePriceActiveNext>,
+) -> bool {
+    if let (Some(active), Some(next)) = (active, next) {
+        let active_price = match Decimal::from_str_exact(&active.amount) {
+            Ok(num) => num,
+            Err(_) => return false,
+        };
+
+        let next_price = match Decimal::from_str_exact(&next.amount) {
+            Ok(num) => num,
+            Err(_) => return false,
+        };
+
+        if active_price <= Decimal::zero() || next_price <= Decimal::zero() {
+            return false;
+        }
+
+        let deviation_threshold = Decimal::new(5, 1); // This represents 0.5
+
+        let diff = next_price - active_price;
+        let abs_diff = diff.abs();
+        let threshold = active_price * deviation_threshold;
+        if abs_diff >= threshold {
+            return false;
+        }
+
+        true
+    } else {
+        false
+    }
```

Comment:
We can return a Result<bool> here instead.

```suggestion
    let (Some(active), Some(next)) = (active, next) else { return Ok(false) };
    
    let active_price = Decimal::from_str_exact(&active.amount)?;
    let next_price = Decimal::from_str_exact(&next.amount)?;

    if active_price <= Decimal::zero() || next_price <= Decimal::zero() {
        return Ok(false);
    }

    let deviation_threshold = Decimal::new(5, 1); // This represents 0.5

    let diff = next_price - active_price;
    let abs_diff = diff.abs();
    let threshold = active_price * deviation_threshold;
    if abs_diff >= threshold {
        return Ok(false);
    }

    Ok(true)
```

We don't want to silence the error. An error here doesn't mean that the price is not live but that there was an error parsing the amounts